controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -239,14 +241,21 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 	} else if r.StorageClassRequest.Spec.Type == "sharedfilesystem" {
 		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
 		r.cephFilesystemSubVolumeGroup.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephFilesystemSubVolumeGroup" {
-				r.cephFilesystemSubVolumeGroup.Name = res.Name
-				break
-			}
+
+		cephFilesystemSubVolumeGroupList := &rookCephv1.CephFilesystemSubVolumeGroupList{}
+		listOptions := &client.MatchingLabels{
+			controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
+			controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
```

Comment:
Why do we care about this? we are fetching by owner anyways. What I do think we should filter is by namespace.